spring-integration-core/src/main/java/org/springframework/integration/support/utils/IntegrationUtils.java

```diff
@@ -193,4 +195,18 @@ public static RuntimeException wrapInHandlingExceptionIfNecessary(Message<?> mes
 		return runtimeException;
 	}
 
+	/**
+	 * Obtain a component name from the provided {@link NamedComponent}.
+	 * @param component the {@link NamedComponent} source for component name.
+	 * @return the component name
+	 * @since 5.3
+	 */
+	public static String obtainComponentName(NamedComponent component) {
+		String name = component.getComponentName();
+		if (name.startsWith('_' + IntegrationConfigUtils.BASE_PACKAGE)) {
```

Comment:
This would be more efficient as `if (name.charAt(0) == '_' && name.startsWith(...`.

Not a big deal with JMX but we're building the graph more often than during initialization.